src/main/kotlin/stringLettersCalculator/StringCalculator.kt

```diff
@@ -0,0 +1,38 @@
+package stringLettersCalculator
+
+
+class StringCalculator {
+
+    fun calculate(inputMessage: String): Int {
+        val splitKeyWord = splitKeyWord(inputMessage)
+        return StringNumbers.from(splitKeyWord)
+            .sumNumbers()
+    }
+
+    private fun splitKeyWord(inputMessage: String): List<String> {
+        val delimiters = mutableListOf(",", ":")
+        val result = Regex(CUSTOM_REGEX_PATTERN).find(inputMessage)
+
+        result?.let { it ->
+            val customDelimiter = it.groupValues[1]
+            delimiters.add("\\" + customDelimiter)
+
+            return splitByDelimiters(it.groupValues[2], delimiters)
+        }
+        return splitByDelimiters(inputMessage, delimiters)
```

Comment:
let 스코프 함수를 사용해볼 수도 있지만,
코틀린의 if는 "문"이 아닌 "식"이기 때문에 아래 처럼도 활용해볼 수 있어요.
취향의 영역이라 생각해서, 참고정도만 해주셔도 좋습니다 :)
```suggestion
      return if (result == null) {
          splitByDelimiters(inputMessage, delimiters)
      } else {
          val customDelimiter = it.groupValues[1]
          delimiters.add("\\" + customDelimiter)

          splitByDelimiters(it.groupValues[2], delimiters)
      }
```

Reply:
아 이건 제가 약간 코틀린스럽게? 해보고 싶은 욕심이 과해서 분기문 대신 저런식으로 표현을 해봤습니다. 의견 충분히 공감합니다!